homeassistant/components/yessssms/notify.py

```diff
@@ -8,34 +8,49 @@
 
 from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
 
+from YesssSMS import YesssSMS
+
+from .const import CONF_PROVIDER
+
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_RECIPIENT): cv.string,
+        vol.Optional(CONF_PROVIDER, default="YESSS"): cv.string,
     }
 )
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the YesssSMS notification service."""
-    return YesssSMSNotificationService(
-        config[CONF_USERNAME], config[CONF_PASSWORD], config[CONF_RECIPIENT]
+
+    try:
+        yesss = YesssSMS(
+            config[CONF_USERNAME], config[CONF_PASSWORD], provider=config[CONF_PROVIDER]
+        )
+    except (KeyError, YesssSMS.UnsupportedProviderError) as ex:
+        _LOGGER.error("Unknown provider: %s", ex)
+        return None
+    except YesssSMS.MissingLoginCredentialsError as ex:
```

Comment:
This error should be impossible since we have validated that the credentials are passed in.

It would be good to instead check that the credentials are correct and we can login.

Reply:
I agree, but I'm not sure if we should 'leak' every reboot of Home-Assistant to the provider.

any suggestion?

Reply:
maybe a `test_login` flag in the config?

Reply:
It's standard procedure to check that credentials are correct when setting up home assistant. All integrations should do this.

Is there a special security concern with these providers?

Reply:
I don't think so. It's just that I want to be as privacy conscious as possible. At least there should be a flag to disable it, if it's on by default. 
Also there is no need to check on every restart of HA if you know it works.

The questions is should it be on by default - I guess yes. so I'll add a flag to disable it.

Reply:
We should not add a config option. This is standard procedure. Please add the check.

Reply:
If this integration is reworked to use config entries the check will only be necessary when creating the config entry.